pandas/core/algorithms.py

```diff
@@ -440,7 +440,12 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
     # GH16012
     # Ensure np.in1d doesn't get object types or it *may* throw an exception
     if len(comps) > 1_000_000 and not is_object_dtype(comps):
```

Comment:
Sorry a little bit too late. The whole point of doing this for len(comps) > 1_000_000, was that numpy was deemed to be faster (which is probably no loner the case btw, see https://github.com/pandas-dev/pandas/issues/22205#issuecomment-410519409), adding `any`, `isnan`, `logical_or` on top  (with all the cache misses and temporary objects) will make this branch much slower. So probably it is best just to drop the whole branch and always keep `f = htable.ismember_object ` (unless it is `is_integer_dtype` of cause).

Reply:
can u run the asvs and check here?

Reply:
@jreback I have opened RP #36611 with my suggestion and some benchmarks, which show that numpy's `in1d` is only faster when here are very few  unique `values`.